### PR TITLE
Allow passing client context to recipe service client methods

### PR DESCRIPTION
**Pull request by ctrombley**
Depends on newrelic/newrelic-client-go#563

### CODE REVIEW

File: internal/install/install.go [hunk 1]
```diff
@@ -119,16 +117,16 @@ func executeRecipeSteps(ctx context.Context, r recipeFile) error {
...
-	ch := make(chan os.Signal, 1)
-	signal.Notify(ch, os.Interrupt, syscall.SIGTERM)
-	ctx, cancel := context.WithCancel(context.Background())
-	go func() {
-		sig := <-ch
-		log.Warnf("signal received: %s", sig)
-		cancel()
-	}()
-	return ctx
+func setSystemVars(t *taskfile.Taskfile, m discoveryManifest) {
```
Comment (zlesnr) [inline internal/install/install.go:120]: 👍

File: internal/install/service_recipe_fetcher.go [hunk 2]
```diff
@@ -30,21 +30,19 @@ func (f *serviceRecipeFetcher) fetchRecommendations(manifest *discoveryManifest)
...
 
-func (f *serviceRecipeFetcher) fetchFilters() ([]recipeFilter, error) {
+func (f *serviceRecipeFetcher) fetchFilters(ctx context.Context) ([]recipeFilter, error) {
 	var resp recipeFilterQueryResult
-	if err := f.client.QueryWithResponse(recipeFilterCriteriaQuery, nil, &resp); err != nil {
+	if err := f.client.QueryWithResponseAndContext(ctx, recipeFilterCriteriaQuery, nil, &resp); err != nil {
 		return nil, err
 	}
 
-	fmt.Printf("resp: %+v\n", resp)
```
Comment (zlesnr) [inline internal/install/service_recipe_fetcher.go:46]: 👍

File: internal/install/psutil_discoverer.go [hunk 3]
```diff
@@ -53,10 +53,11 @@ func (p *psUtilDiscoverer) discover(ctx context.Context) (*discoveryManifest, er
 		processes = append(processes, psUtilProcess(*pp))
 	}
 
-	filtered, err := p.processFilterer.filter(processes)
-	if err != nil {
-		return nil, err
-	}
+	filtered := processes
```
Comment (zlesnr) [inline internal/install/psutil_discoverer.go:56]: No filtering for now?

**Review by zlesnr — DISMISSED** [review]

**Review by zlesnr — APPROVED** [review]

File: internal/install/psutil_discoverer.go (hunk 3 above)
Comment (ctrombley) [reply internal/install/psutil_discoverer.go:56]: The recipe service hasn't exposed the filters yet to be able to drive this.

**Pull request merged**